src/backend/utils/misc/guc.c

```diff
@@ -10090,11 +10090,11 @@ assign_effective_io_concurrency(int newval, void *extra)
 static bool
 check_client_connection_check_interval(int *newval, void **extra, GucSource source)
 {
-#ifndef POLLRDHUP
-	/* Linux only, for now.  See pq_check_connection(). */
+#if !(defined(POLLRDHUP) || defined(__darwin__))
+	/* Linux and OSX only, for now.  See pq_check_connection(). */
 	if (*newval != 0)
 	{
-		GUC_check_errdetail("client_connection_check_interval must be set to 0 on platforms that lack POLLRDHUP.");
+		GUC_check_errdetail("client_connection_check_interval must be set to 0 on platforms that lack POLLRDHUP and not OSX.";
```

Comment:
Note this check is necessary for other parts of this commit to be safe on systems other than having `POLLRDHUP` and OSX.